completion.bash

```diff
@@ -0,0 +1,22 @@
+# TODO(michael.cusack): Currently just completes the first arg (command name). Update to also support command args.
+_clr_completion()
+{
+    local IFS=$'\n'
+
+    case $COMP_CWORD in
+        1)
+            # First argument. Completes the namespace:command.
+            _get_comp_words_by_ref -n : cur
```

Comment:
The default shell on OS X is now zsh, and I know much of the team are zsh users.

Even when I'm in a `bash` shell (on Ubuntu) and have the `bash-completion` package installed, I get errors.

```
$ clr d<tab literal>bash: _get_comp_words_by_ref: command not found
bash: clr: command not found
bash: __ltrim_colon_completions: command not found
```

I guess this isn't a huge deal, since people need to opt in to shell completion, but I'm just flagging.

Reply:
Sorry, the `_get_comp_words_by_ref` failure was 100% just due to my rc files not sourcing `/etc/profile.d/bash_completion.sh` properly. That's my bad: I can get this working in `bash`.

The question still stands how we'd support zsh users, if we care to. :)

Reply:
Targeting zsh and bash eventually. Starting with bash.